### ISSUE TITLE
Feasibility question: migrating from a setup with generated Dockerfiles?

### ISSUE THREAD

**Issue by rulatir** [earthly/earthly#615]
Earthly "annexes" the realm of image building into the realm of "build-systemming" (for a lack of better term). Is it possible to support a scenario where (under the build system we currently use) some Dockerfiles are generated as part of the build process, and only then are images built from them? Since Earthfiles replace Dockerfiles, we would have to be generating (some) Earthfiles. This effectively places a requirement on Earthly to be able to handle self-modifying build specs: there will be Earthfiles that must themselves be built/rebuilt as part of the build process. Does Earthly satisfy this requirement?

**Comment by vladaionescu** [comment]
Hi @rulatir - yes, Earthly fits this use-case. You can definitely use generated Dockerfiles within Earthly. Here is a simple example:

```
generate:
    FROM ...
    COPY ...
    RUN whatever-command-you-use-to-generate-Dockerfile ...
    SAVE ARTIFACT ./Dockerfile
    SAVE ARTIFACT ./the-rest-of-the-files-that-docker-build-typically-needs

build-dockerfile:
    FROM DOCKERFILE +generate/*
    SAVE IMAGE myimage:latest
```

We've seen this approach be popular with users of `sbt`.

For more info on `FROM DOCKERFILE` see the [reference doc](https://docs.earthly.dev/earthfile#from-dockerfile-beta).

**Comment by rulatir** [comment]
Thanks! I looked into the docs, and sadly current limitations of FROM DOCKERFILE render it unsuitable for my purpose due to lack of `-f` support and the assumption that context dir is the same as the dir containing the dockerfile. One requirement that I set for a build system is that it must handle the current organization of the monorepo, without me having to move things around to put them where build system expects them. Instead it must be completely 100% possible to imperatively tell the build system where things are. I will check again once FROM LOCAL is implemented, which will probably just let me run `docker build`.
